### DISCUSSION TITLE
web3Modal not working properly on android metamask

### DISCUSSION THREAD

**Discussion by agam-w** [WalletConnect/walletconnect-monorepo#3261] [Developer Support]
![WhatsApp Image 2023-07-25 at 9 26 31 AM](https://github.com/WalletConnect/walletconnect-monorepo/assets/30470275/ff685a6d-46d3-4cc5-a41f-5c3e96f44027)

https://github.com/WalletConnect/walletconnect-monorepo/assets/30470275/9224bd33-bc90-4379-a9ae-06a106c836c2

Hi WalletConnect team, i found and issue on web3modal when open it with metamask mobile android, first i found this issue on our platform but when i check https://web3modal.com/ it show same issue with our platform.

so you can see on the video is when i try to connect using metamask android there will be some issue:
1. the recomended wallet not showing (can see the image i attached is from iOS).
2. when click connect with metamask will show alert “This website has been blocked from automatically opening an external application” with “IGNORE” and “ALLOW” options button. even its already on metamask.
3. after click the "ALLOW" and connecting it will close the metamask app
4. it will logged in after you open the metamask again but with wrong network (can see the video it still on arbitrum)

realy appreciate for WalletConnect team if you guys can help me on this issue, thanks.

FYR here library used on our web base platform:
"viem": "^1.4.1",
"wagmi": "^1.3.9",
"@web3modal/ethereum": "^2.7.0",
"@web3modal/react": "^2.7.0",

**Comment by glitch-txs** [comment]
Tested on Android and MetaMask icon is showing on my end. You're using the MetaMask explorer right? Which Android are you testing with?

Could you test on labs?
https://lab.web3modal.com/

**Reply by agam-w** [reply]
@glitch-txs thanks for your reply, yes im using Metamask app browser latest version v.7.3.0. i already test using 2 device:
1. samsung s20 android version 13
2. redmi note 10S android version 13

both using latest metamask version v.7.3.0

https://lab.web3modal.com/ already try this and it works, can you give me example for this code please.

since i already build my code with the docs given and seems not working on android and show like the video (take from https://web3modal.com)

**Reply by glitch-txs** [reply]
You can find labs code in the web3modal repo. 

I was able to replicate in the main website after switching chain. Is this issue MetaMask only?

**Reply by agam-w** [reply]
@glitch-txs  thanks will look into it, i only try using metamask for now and to be spesific the issue is only happened on android metamask app. it works perfectly on iOS metamask app.
